app/models/solidus_stripe/customer.rb

```diff
@@ -15,7 +15,21 @@ def self.retrieve_or_create_stripe_customer_id(payment_method:, order:)
     end
 
     def create_stripe_customer
-      payment_method.gateway.request { Stripe::Customer.create(email: source.email) }
+      payment_method.gateway.request { Stripe::Customer.create(email: source.email, name: customer_name) }
+    end
+
+    def customer_name
+      return full_name if full_name.present?
+
+      source.email
```

Comment:
This can be written simply as `full_name.presence || source.email` if you prefer.